client/lib/catch-js-errors/index.js

```diff
@@ -20,80 +13,18 @@
 		}
 	}
 
-	function sendErrorsToApi() {
-		var xhr = new XMLHttpRequest(),
-			url = ( window.location && window.location.href ) ? window.location.href + ' ' : '',
-			params;
-
-		if ( savedErrors.length > 0 ) {
-			// POST to the API
-			xhr.open( 'POST', 'https://public-api.wordpress.com/rest/v1.1/js-error?http_envelope=1', true );
-			xhr.setRequestHeader( 'Content-type', 'application/x-www-form-urlencoded' );
-
-			params = 'client_id=39911&client_secret=cOaYKdrkgXz8xY7aysv4fU6wL6sK5J8a6ojReEIAPwggsznj4Cb6mW0nffTxtYT8&error=';
-			params += encodeURIComponent( url + JSON.stringify( savedErrors ) );
-			//Window.location.href is hached in here, because backend has a char limit for now.
-
-			xhr.send( params );
-
-			savedErrors = [];
-		}
-	}
-
-	function errorToPlainObject( error ) {
-		var simpleObject = {};
-		Object.getOwnPropertyNames( error ).forEach( function( key ) {
-			simpleObject[ key ] = error[ key ];
-		} );
-		return simpleObject;
-	}
-
-	function handleError( error ) {
-		var canSendNow;
-
-		// if we have packed too much messages, it's probably good to stop stacking more and drop new messages while the buffer is flushed
-		if ( savedErrors.length > MAX_BUFFER_SIZE ) {
-			return;
-		}
-
-		// we can send a message we have waited long enough (because we are throttling)
-		canSendNow = +new Date() - lastTimeSent > THROTTLE_DURATION;
-
-		// add the message to the pack and reset flush timeout
-		clearTimeout( packTimeout );
-		savedErrors.push( errorToPlainObject( error ) );
-
-		// if we can send the pack now, let's do it
-		if ( canSendNow ) {
-			flushErrors();
-		} else {
-			// else set a timeout to ensure message is sent at least THROTTLE_DURATION after it happened
-			// This is needed as a message could otherwise stay in the pack forever if no message comes afterwards
-			packTimeout = setTimeout( function() {
-				flushErrors();
-			}, THROTTLE_DURATION );
-		}
-	}
-
-	function flushErrors() {
-		lastTimeSent = +new Date();
-		sendErrorsToApi();
-	}
-
-	function saveError( message, scriptUrl, lineNumber, columnNumber, error ) {
-		error = error || new Error( message );
-
-		// Add user agent if we have it
-		if ( navigator && navigator.userAgent ) {
-			error.userAgent = navigator.userAgent;
-		}
-
-		handleError( error );
+	function sendToApi( error ) {
+		const xhr = new XMLHttpRequest();
+		xhr.open( 'POST', 'https://public-api.wordpress.com/rest/v1.1/js-error?http_envelope=1', true );
+		xhr.setRequestHeader( 'Content-type', 'application/x-www-form-urlencoded' );
+		let params = 'client_id=39911&client_secret=cOaYKdrkgXz8xY7aysv4fU6wL6sK5J8a6ojReEIAPwggsznj4Cb6mW0nffTxtYT8&error=';
```

Comment:
This key is already public in our config files https://github.com/Automattic/wp-calypso/blob/master/config/development.json#L13
It seems to belong to a standard OAuth2 app created for signup in wordpress.com, it doesn't have any special rights.